From a10a2cb2f04ff6ca09d84d56c7bc9b4ce0266c76 Mon Sep 17 00:00:00 2001
From: Jiri Pirko <jpirko@redhat.com>
Date: Thu, 7 Aug 2014 07:06:34 -0400
Subject: [net] vti4: Don't count header length twice

Message-id: <1407395194-3809-19-git-send-email-jpirko@redhat.com>
Patchwork-id: 87531
O-Subject: [patch rhel7 net 18/18] vti4: Don't count header length twice.
Bugzilla: 1091561
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

BZ1091561
https://brewweb.devel.redhat.com/taskinfo?taskID=7794261
Tested by jtluka's LNST smoke test recipe

upstream commit a32452366b7250c42e96a18ffc3ad8db9e0ca3c2
Author: Steffen Klassert <steffen.klassert@secunet.com>
Date:   Wed Apr 16 09:01:03 2014 +0200

    vti4: Don't count header length twice.

    We currently count the size of LL_MAX_HEADER and struct iphdr
    twice for vti4 devices, this leads to a wrong device mtu.
    The size of LL_MAX_HEADER and struct iphdr is already counted in
    ip_tunnel_bind_dev(), so don't do it again in vti_tunnel_init().

    Fixes: b9959fd3 ("vti: switch to new ip tunnel code")

Signed-off-by: Jiri Pirko <jpirko@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv4/ip_vti.c b/net/ipv4/ip_vti.c
index 8fe4bdd..b6d7f97 100644
--- a/net/ipv4/ip_vti.c
+++ b/net/ipv4/ip_vti.c
@@ -349,7 +349,6 @@ static int vti_tunnel_init(struct net_device *dev)
  memcpy(dev->dev_addr, &iph->saddr, 4);
  memcpy(dev->broadcast, &iph->daddr, 4);
 
- dev->hard_header_len = LL_MAX_HEADER + sizeof(struct iphdr);
  dev->mtu  = ETH_DATA_LEN;
  dev->flags  = IFF_NOARP;
  dev->iflink  = 0;
-- 
1.7.1